File: doc-examples/example-java/src/main/java/io/micronaut/gcp/pubsub/quickstart/AnimalService.java
package io.micronaut.gcp.pubsub.quickstart;
//tag imports[]
import io.micronaut.gcp.pubsub.support.Animal;
import javax.inject.Singleton;
// end::imports[]

// tag::clazz[]
@Singleton
public class AnimalService {
    private final AnimalClient animalClient;

    public AnimalService(AnimalClient animalClient) { // <1>
        this.animalClient = animalClient;
    }

    public void someBusinessMethod(Animal animal) {
        byte[] serializedBody = serialize(animal);
        animalClient.send(serializedBody);
    }

    private byte[] serialize(Animal animal) { // <2>
        return null;
    }
}
// end::clazz[]

File: doc-examples/example-kotlin/src/main/kotlin/io/micronaut/gcp/pubsub/quickstart/AnimalService.kt
package io.micronaut.gcp.pubsub.quickstart

//tag imports[]
import io.micronaut.gcp.pubsub.support.Animal
import javax.inject.Singleton
// end::imports[]

// tag::clazz[]
@Singleton
class AnimalService(private val animalClient: AnimalClient)  { // <1>

	fun someBusinessMethod(animal: Animal) {
		val serializedBody = serialize(animal)
		animalClient.send(serializedBody)
	}

	private fun serialize(animal: Animal): ByteArray { // <2>
		return ByteArray(0)
	}

}
// end::clazz[]